packages/teams-js/src/private/externalAppAuthentication.ts

```diff
@@ -0,0 +1,329 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { authentication, FrameContexts } from '../public';
+import { errorNotSupportedOnPlatform } from '../public/constants';
+import { runtime } from '../public/runtime';
+
+/**
+ * @hidden
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export namespace externalAppAuthentication {
+  /*********** BEGIN REQUEST TYPE ************/
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type OriginalRequestInfo = IQueryMessageExtensionRequest | IActionExecuteInvokeRequest;
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface IQueryMessageExtensionRequest {
+    requestType: OriginalRequestType.QueryMessageExtensionRequest;
+    commandId: string;
+    parameters?: {
+      name?: string;
+      value?: string;
+    }[];
+    queryOptions?: {
+      count: number;
+      skip: number;
+    };
+  }
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  interface IActionExecuteInvokeRequest {
+    requestType: OriginalRequestType.ActionExecuteInvokeRequest;
+    type: string; // "invoke"
```

Comment:
>string [](http://example.com/codeflow?start=10&length=6)

Is type really a free string? There's no enumeration of possible types?

Reply:
https://adaptivecards.io/explorer/Action.Execute.html - can only be "Action.Execute". we can add validation for this.

Reply:
Added validation